src/pages/settings/Subscription/CardSection/utils.ts

```diff
@@ -1,9 +1,140 @@
+            return {
+                title: translate('subscription.billingBanner.outdatedInfo'),
+                subtitle: translate('subscription.billingBanner.paymentPastDue'),
+                isError: true,
+                isAddButtonDark: true,
+            };
+
+        case SubscriptionUtils.PAYMENT_STATUSES.BILLING_DISPUTE_PENDING:
+            return {
+                title: translate('subscription.billingBanner.cardCouldNotBeCharged'),
+                subtitle: translate('subscription.billingBanner.cardOnDispute', {amountOwed, cardEnding}),
+                isError: true,
+                isRetryAvailable: false,
+            };
+
+        case SubscriptionUtils.PAYMENT_STATUSES.CARD_AUTHENTICATION_REQUIRED:
+            return {
+                title: translate('subscription.billingBanner.cardCouldNotBeCharged'),
+                subtitle: translate('subscription.billingBanner.cardNotFullyAuthenticated', {cardEnding}),
+                isError: true,
+                isAuthenticatingRequired: true,
+            };
+
+        case SubscriptionUtils.PAYMENT_STATUSES.INSUFFICIENT_FUNDS:
+            return {
+                title: translate('subscription.billingBanner.cardCouldNotBeCharged'),
+                subtitle: translate('subscription.billingBanner.cardDeclinedDueToInsufficientFunds', {amountOwed}),
+                isError: true,
+                isRetryAvailable: true,
+            };
+
+        case SubscriptionUtils.PAYMENT_STATUSES.CARD_EXPIRED:
+            return {
+                title: translate('subscription.billingBanner.cardCouldNotBeCharged'),
+                subtitle: translate('subscription.billingBanner.cardExpired', {amountOwed}),
+                isError: true,
+                isRetryAvailable: true,
+            };
+
+        case SubscriptionUtils.PAYMENT_STATUSES.CARD_EXPIRE_SOON:
+            return {
+                title: translate('subscription.billingBanner.cardExpiringSoon'),
+                subtitle: translate('subscription.billingBanner.cardWillExpireAtTheEndOfMonth'),
+                isError: false,
+                icon: Illustrations.CreditCardEyes,
+            };
+
+        case SubscriptionUtils.PAYMENT_STATUSES.RETRY_BILLING_SUCCESS:
+            return {
+                title: translate('subscription.billingBanner.succeeded'),
+                subtitle: translate('subscription.billingBanner.billedSuccessfully'),
+                isError: false,
+                rightIcon: Expensicons.Close,
+            };
+
+        case SubscriptionUtils.PAYMENT_STATUSES.RETRY_BILLING_ERROR:
+            return {
+                title: translate('subscription.billingBanner.cardCouldNotBeCharged'),
+                subtitle: translate('subscription.billingBanner.retryMessage'),
+                isError: true,
+            };
+
+        default:
+            return {};
+    }
+}
+
+function getCardForSubscriptionBilling(): Fund | undefined {
```

Comment:
Do we need this extra method? Why not call `SubscriptionUtils` from the component?

Reply:
Sure